python/gpusim_server.py

```diff
@@ -73,16 +80,33 @@ def do_GET(self):
         self.send_error(404, 'Server unavailable.')
 
     def get_data_from_dbname(self, dbname):
-        if dbname not in sockets:
-            self.send_error(404, f'DB {dbname} not found in options: {sockets.keys()}') #noqa
-            raise ValueError('DB not found')
-        allsmiles, allids, allscores = [], [], []
-        src_smiles, return_count, similarity_cutoff = self.get_posted_data()
-        output_qba = self.get_data(dbname, src_smiles, return_count,
-                                    similarity_cutoff)
-        return_count, smiles, ids, scores = self.deserialize_results(output_qba)
-
-        return smiles, ids, scores, src_smiles
+        global search_mutex
+        search_mutex.lock()
+        try:
+            if dbname not in sockets:
+                self.send_error(404, f'DB {dbname} not found in options: {sockets.keys()}') #noqa
+                raise ValueError('DB not found')
+            allsmiles, allids, allscores = [], [], []
+            src_smiles, return_count, similarity_cutoff = self.get_posted_data()
+            request_num = random.randint(0, 2**31)
+            print("Processing request {}".format(request_num), file=sys.stderr)
+            output_qba = self.get_data(dbname, src_smiles, return_count,
+                                        similarity_cutoff, request_num)
+            try:
+                return_count, smiles, ids, scores = self.deserialize_results(
+                    request_num, output_qba)
+            except RuntimeError:
+                self.flush_socket(dbname)
+                raise
+
+            return smiles, ids, scores, src_smiles
+        finally:
+            search_mutex.unlock()
+
+    def flush_socket(self, dbname):
```

Comment:
what is dbname being used for?

Reply:
Well...this shows we haven't hit this code yet, as theoretically it should be being used where socket_name is...

Reply:
Fixed